### PR TITLE
refactor(search): use context

### PR DESCRIPTION
**Pull request by hengkx**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by hengkx** [comment]
奇怪怎么另外一个 文件 覆盖率 降低了

**Comment by hengkx** [comment]
wait #27694

File: components/input/Search.tsx [hunk 1]
```diff
@@ -22,10 +22,29 @@ export interface SearchProps extends InputProps {
...
+    addonAfter,
+    loading,
+    disabled,
+    onSearch: customOnSearch,
+    onChange: customOnChange,
+    ...restProps
+  } = props;
+
+  const { getPrefixCls, direction } = React.useContext(ConfigContext);
+  const contextSize = React.useContext(SizeContext);
```
Comment (zombieJ) [inline components/input/Search.tsx:41]: 超过 2 个相同 use 的时候改成 `import { useXXX } from 'antd';`，来减少 bundle size

File: components/input/Search.tsx [hunk 2]
```diff
@@ -41,14 +60,15 @@ const Search = React.forwardRef<Input, SearchProps>((props, ref) => {
...
       customOnSearch(inputRef.current?.input.value!, e);
     }
   };
 
-  const renderAddonAfter = (prefixCls: string, size: SizeType) => {
-    const { enterButton, disabled, addonAfter, loading } = props;
+  const prefixCls = getPrefixCls('input-search', customizePrefixCls);
+  const inputPrefixCls = getPrefixCls('input', customizeInputPrefixCls);
+
+  const renderAddonAfter = () => {
```
Comment (zombieJ) [inline components/input/Search.tsx:71]: 看起来这个方法也没用了

**Review by zombieJ — APPROVED** [review]

File: components/input/Search.tsx (hunk 1 above)
Comment (hengkx) [reply components/input/Search.tsx:41]: SizeContext 和 ConfigContext 组合成一个 hook？

File: components/input/Search.tsx (hunk 2 above)
Comment (hengkx) [reply components/input/Search.tsx:71]: 看了一眼 里面 代码 都是 他自己 使用的 ，去不掉

Comment (zombieJ) [reply components/input/Search.tsx:71]: 我意思是不需做成方法了，可以直接打平~

**Pull request merged**